quantestpy/simulator/pauli_circuit.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+
+from quantestpy.exceptions import QuantestPyError, QuantestPyTestCircuitError
+from quantestpy.test_circuit import TestCircuit
+
+_IMPLEMENTED_GATES = ["x", "y", "z", "swap"]
+
+
+class PauliCircuit(TestCircuit):
+
+    def __init__(self, num_qubit: int):
+        if not isinstance(num_qubit, int) or num_qubit < 1:
+            raise QuantestPyTestCircuitError(
+                "num_qubit must be an integer greater than 0."
+            )
+
+        super().__init__(num_qubit=num_qubit)
+        self._qubit_value = np.array([0 for _ in range(num_qubit)])
+        self._qubit_phase = np.array([0. for _ in range(num_qubit)])
+
+    def add_gate(self, gate: dict) -> None:
+        if not isinstance(gate, dict):
```

Comment:
I think it is bettter to use the same message as TestCircuit.

'''
        if not isinstance(gate, dict):
            raise QuantestPyTestCircuitError(
                "gate's type must be dictionary which contains 'name', "
                "'target_qubit' and optionally 'control_qubit' as keys."
            )
'''